core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -169,8 +169,250 @@ public void testJoiner() throws Exception {
     DataSetManager<TimePartitionedFileSet> outputManager = getDataset(joinedDatasetName);
     TimePartitionedFileSet fileSet = outputManager.get();
 
+    // verfiy innerjoin output
+    verifyInnerJoinOutput(outputSchema, fileSet);
+  }
+
+  @Test
+  public void testOuterJoin() throws Exception {
```

Comment:
these tests look mostly the same except for what is verified at the end and the join configuration. Could you refactor so that they share whatever code is the same?